reference-target-explainer.md

```diff
@@ -0,0 +1,611 @@
+- `shadowrootfortarget`
+- `shadowrootformtarget`
+- `shadowrootlisttarget`
+- `shadowrootpopovertargettarget`
+- `shadowrootinvoketargettarget`
+- `shadowrootinteresttargettarget`
+- `shadowrootheaderstarget`
+- `shadowrootitemreftarget`
+- `shadowrootreferencetarget` -- all other references except the ones specified above
+
+Reflected by JavaScript attributes `ShadowRoot.ariaActiveDescendantTarget`, etc.
+
+#### Pros
+
+- Syntax is more in line with other HTML attributes, rather than using a comma-separated list of colon-separated map entries.
+- Works with IDs that contain commas.
+- It is possible to scope support for properties where this behavior has a real use-case, such as `aria-activedescendant`. This would limit the number of new properties to only a handful.
+
+#### Cons
+
+- Adds 15+ new attributes instead of 2.
+- Less clear(?) that `shadowrootreferencetarget` only forwards references that are not explicitly specified by other elements.
+
+### Designate target elements using attributes instead of IDREF
+
+The [Cross-root ARIA Reflection API](https://github.com/Westbrook/cross-root-aria-reflection/blob/main/cross-root-aria-reflection.md#usage-with-declarative-shadow-dom) explainer proposes adding attributes to elements inside the shadow tree:
+
+```html
+<x-foo id="foo">
+  <template shadowroot="open" shadowrootreflectscontrols shadowrootreflectsariaactivedescendent>
+    <ul reflectariacontrols>
+      <li>Item 1</li>
+      <li reflectariaactivedescendent>Item 2</li>
+      <li>Item 3</li>
+    </ul>
+  </template>
+</x-foo>
+```
+
+#### Pros
+
+- Does not require an ID on the target element. [But does still require an extra attribute; possibly in addition to an ID if that ID is used for other purposes.]
+
+#### Cons
+
+- Requires new attributes in two places in order to work: E.g. `shadowrootreflectscontrols` on the shadow root _and_ `reflectariacontrols` on the target element.
+- When multiple elements are used for the same attribute, the author cannot control the order (the order is always the DOM order).
+
+### Use exported IDs instead of per-attribute targeting
+
+The [ExportID explainer](https://github.com/WICG/aom/blob/gh-pages/exportid-explainer.md) proposes a way to refer to elements inside the shadow DOM by name. For example, `"fancy-input::id(real-input)"` to refer to a specific `<input>` element inside a `<fancy-input>`.
+
+It would be possible to use exported IDs instead of `referenceTargetMap` if/when it is necessary to refer to an element other than the primary reference target.
+
+#### Pros
+
+- Does not suffer from the [bottleneck effect](https://alice.pages.igalia.com/blog/how-shadow-dom-and-accessibility-are-in-conflict/#limitations-of-these-apis).
+- Potentially less confusing why you reference the _container_ listbox with `aria-activedescendant` instead of the element itself.
+
+#### Cons
+
+- Exposes some of the internal details of a control and does not give a way for the control to encapsulate those details.
```

Comment:
I'm still not really clear on how this is the case any more than it is for `::part()`.

Reply:
I added a sub-bullet to this con to capture the feedback:
>   * This may not be a dealbreaker: the [`::part()`](https://drafts.csswg.org/css-shadow-parts/#part) CSS selector also has a similar drawback for CSS styles, but it still is a standard and a useful feature for styles.